python/kvikio/nvcomp.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
+
+from enum import Enum
+
+import cupy as cp
+import numpy as np
+
+import kvikio._lib.pynvcomp as _lib
+
+
+def cp_to_nvcomp_dtype(in_type: cp.dtype) -> Enum:
+    """ Convert np/cp dtypes to nvcomp integral dtypes.
+
+    Returns
+    -------
+    int
+        The value of the NVCOMP_TYPE for supported dtype.
+    """
+    cp_type = cp.dtype(in_type)
+    return {
+        cp.dtype("int8"): _lib.pyNvcompType_t.pyNVCOMP_TYPE_CHAR,
+        cp.dtype("uint8"): _lib.pyNvcompType_t.pyNVCOMP_TYPE_UCHAR,
+        cp.dtype("int16"): _lib.pyNvcompType_t.pyNVCOMP_TYPE_SHORT,
+        cp.dtype("uint16"): _lib.pyNvcompType_t.pyNVCOMP_TYPE_USHORT,
+        cp.dtype("int32"): _lib.pyNvcompType_t.pyNVCOMP_TYPE_INT,
+        cp.dtype("uint32"): _lib.pyNvcompType_t.pyNVCOMP_TYPE_UINT,
+        cp.dtype("int64"): _lib.pyNvcompType_t.pyNVCOMP_TYPE_LONGLONG,
+        cp.dtype("uint64"): _lib.pyNvcompType_t.pyNVCOMP_TYPE_ULONGLONG,
+    }[cp_type]
+
+
+class CascadedOptions:
```

Comment:
I felt like this configuration made the parameters the least visible to end users, who I presume have no use for them unless they're power users with very specific data distributions that could profit from tweaking these parameters. Having anything in the default constructor signature other than "CascadedOptions" makes it seem like the user of `CascadedCompressor` needs to know about them. That's my theory, anyway, it could be more cognitive load as you suggest.

Reply:
@madsbk @jakirkham what do you think? IMO keyword arguments are much more Pythonic than config objects, which I don't expect users to be familiar with. I think users are accustomed to ignoring keyword arguments that they don't understand or need to know about.

Reply:
I agree, keyword arguments with useful default values are more Pythonic. 
I think a `CascadedOptions` would make more sense if there are multiple sets of default options the user should choose between.

Reply:
Ok, I changed it to named defaults.